HotSpot/Models/TimeInterval.m

```diff
@@ -0,0 +1,50 @@
+//
+//  TimeInterval.m
+//  HotSpot
+//
+//  Created by drealin on 7/19/19.
+//  Copyright © 2019 aodemuyi. All rights reserved.
+//
+
+#import "TimeInterval.h"
+
+@interface TimeInterval()<PFSubclassing>
+@end
+
+@implementation TimeInterval
+{
+    NSDateInterval *dateInterval;
+}
+@dynamic startTime;
+@dynamic endTime;
+@dynamic repeatsWeekly;
+
+# pragma mark - Class Methods
++ (nonnull NSString *)parseClassName {
+    return @"TimeInterval";
+}
+
+# pragma mark - Public Methods
+- (NSDateInterval * _Nullable)intersectionWithTimeInterval:(TimeInterval *)timeInterval {
+    if(!dateInterval) {
+        dateInterval = [[NSDateInterval alloc] initWithStartDate:self.startTime endDate:self.endTime];
+    }
+    NSDateInterval *theirDateInterval = [[NSDateInterval alloc] initWithStartDate:timeInterval.startTime endDate:timeInterval.endTime];
+    if (self.repeatsWeekly) {
+        double secondsPerDay = 60 * 60 * 24;
```

Comment:
```suggestion
        CGFloat secondsPerDay = CGFloat(60 * 60 * 24);
```

Reply:
What does CGFloat(60 * 60 * 24) do? Why not just assign secondsPerDAy to 60 * 60 * 24? Actually I get an error with the former, "unexpected type name 'CGFloat': expected expression."